### PR TITLE
fix: next article button on last page

### PR DESCRIPTION
**Pull request by el**
### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/hooks/usePostModalNavigation.ts [hunk 1]
```diff
@@ -117,13 +118,12 @@ export const usePostModalNavigation = (
...
     const firstPost = items.findIndex(isPost);
+    const isLast = items.length - 1 === openedPostIndex;
     if (firstPost === openedPostIndex)
       return items.length - 1 === openedPostIndex
         ? PostPosition.Only
         : PostPosition.First;
-    // We do not set the Last position because the local last might not be the actual last.
-    // The setting can be updated once we update the API to return the information if the page is last or not.
-    return PostPosition.Middle;
+    return !canFetchMore && isLast ? PostPosition.Last : PostPosition.Middle;
```
Comment (el) [inline packages/shared/src/hooks/usePostModalNavigation.ts:126]: Seems like we already had that information.

File: packages/shared/src/hooks/usePostModalNavigation.ts [hunk 2]
```diff
@@ -156,7 +156,7 @@ export const usePostModalNavigation = (
         );
         const item = items[index];
 
-        if (index === items.length) {
+        if (index === items.length && canFetchMore) {
```
Comment (sshanzel) [inline packages/shared/src/hooks/usePostModalNavigation.ts:159]: Nice. Thank you for fixing this as well.

**Review by sshanzel — APPROVED** [review]
Nice work! :shipit:

**Review by sshanzel — CHANGES_REQUESTED** [review]
Oh, I just realized, we should also disable the next button if the `fetchingNextPage` is true, meaning it is loading.

**Review by rebelchris — APPROVED** [review]
Nice one 🤙

File: packages/shared/src/hooks/usePostModalNavigation.ts (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/hooks/usePostModalNavigation.ts:126]: I think we should add the `isFetchingNextPage` here
```suggestion
    return isFetchingNextPage || (!canFetchMore && isLast) ? PostPosition.Last : PostPosition.Middle;
```

Comment (el) [reply packages/shared/src/hooks/usePostModalNavigation.ts]: added

**Review by sshanzel — APPROVED** [review]

**Pull request merged**